pkg/config-disk/config-disk.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package configdisk
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	cloudinit "kubevirt.io/kubevirt/pkg/cloud-init"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/precond"
+)
+
+type ConfigDiskClient interface {
+	Define(vm *v1.VM) error
+	Undefine(vm *v1.VM) error
+}
+
+type configDiskClient struct {
+	unixSocketPath string
+	unixClient     http.Client
+}
+
+func NewConfigDiskClient(unixSocketPath string) ConfigDiskClient {
+	return &configDiskClient{
+		unixSocketPath: unixSocketPath,
+		unixClient: http.Client{
+			Transport: &http.Transport{
+				DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+					return net.Dial("unix", unixSocketPath)
+				},
+			},
+		},
+	}
+}
+
+func (c *configDiskClient) Define(vm *v1.VM) error {
+	namespace := precond.MustNotBeEmpty(vm.GetObjectMeta().GetNamespace())
+	domain := precond.MustNotBeEmpty(vm.GetObjectMeta().GetName())
+
+	if vm.Spec.CloudInit != nil {
+		body, err := json.Marshal(vm.Spec.CloudInit)
+		if err != nil {
+			return err
+		}
+		var response *http.Response
+
+		response, err = c.unixClient.Post("http://virtconfigdisk/create/cloudinit/"+namespace+"/"+domain, "application/json", strings.NewReader(string(body)))
```

Comment:
Issues can occur if domain contains special chars.

Reply:
The domain can't contain special characters.

The pod requires lower case alphanumeric characters with the addition of '-' and '.'  The vm domain will never get started if it's name doesn't match those requirements, which means virt-handler will never see it here.